components/ws-daemon/pkg/iws/iws.go

```diff
@@ -287,7 +287,12 @@ func (wbs *InWorkspaceServiceServer) PrepareForUserNS(ctx context.Context, req *
 		return nil, status.Errorf(codes.Internal, "cannot mount shiftfs mark")
 	}
 
-	if cgroups.Mode() == cgroups.Unified {
+	unified, err := cgroups.IsUnifiedCgroupSetup()
+	if err != nil {
+		return nil, status.Errorf(codes.FailedPrecondition, "could not determine cgroup setup")
```

Comment:
should we include the error `err` in the message?

Reply:
This error might be exposed to the user to whom it would be confusing.
We should log this instead.